### DISCUSSION TITLE
Bonjour je rencontre un erreur

### DISCUSSION THREAD

**Discussion by ghost** [Ariastarcos/mirasaki-music-bot#121] [General]
RIP 2024-2024 PLZ HELP ME

DiscordAPIError[50035]: Invalid Form Body
content[BASE_TYPE_MAX_LENGTH]: Must be 2000 or fewer in length.
    at handleErrors (C:\Users\Darth GXT\Downloads\mirasaki-music-bot-1.1.4\mirasaki-music-bot-1.1.4\node_modules\@discordjs\rest\dist\index.js:722:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async SequentialHandler.runRequest (C:\Users\Darth GXT\Downloads\mirasaki-music-bot-1.1.4\mirasaki-music-bot-1.1.4\node_modules\@discordjs\rest\dist\index.js:1120:23)
    at async SequentialHandler.queueRequest (C:\Users\Darth GXT\Downloads\mirasaki-music-bot-1.1.4\mirasaki-music-bot-1.1.4\node_modules\@discordjs\rest\dist\index.js:953:14)
    at async _REST.request (C:\Users\Darth GXT\Downloads\mirasaki-music-bot-1.1.4\mirasaki-music-bot-1.1.4\node_modules\@discordjs\rest\dist\index.js:1266:22)
    at async InteractionWebhook.editMessage (C:\Users\Darth GXT\Downloads\mirasaki-music-bot-1.1.4\mirasaki-music-bot-1.1.4\node_modules\discord.js\src\structures\Webhook.js:339:15)
    at async ChatInputCommandInteraction.editReply (C:\Users\Darth GXT\Downloads\mirasaki-music-bot-1.1.4\mirasaki-music-bot-1.1.4\node_modules\discord.js\src\structures\interfaces\InteractionResponses.js:158:17) {
  requestBody: {
    files: [],
    json: {
      content: '❌ <@934821383014346772>, something went wrong:\n' +
        '\n' +
        'Could not locate ffmpeg. Tried:\n' +
        '\n' +
        '  1. spawn process.env.FFMPEG_PATH\n' +
        '  2. spawn ffmpeg\n' +
        '  3. spawn avconv\n' +
        '  4. spawn ./ffmpeg\n' +
        '  5. spawn ./avconv\n' +
        '  6. require("@ffmpeg-installer/ffmpeg")\n' +
        '  7. require("ffmpeg-static")\n' +
        '  8. require("@node-ffmpeg/node-ffmpeg-installer")\n' +
        '  9. require("ffmpeg-binaries")\n' +
        '\n' +
        '\n' +
        '=====Full Stacktrace=====\n' +
        '\n' +
        '\n' +
        '\n' +
        '\n' +
        '\n' +
        '\n' +
        '\n' +
        '\n' +
        '\n' +
        'TypeError: Expected spawn process.env.FFMPEG_PATH to be a string, got undefined\n' +
        '    at validatePathParam (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:42:11)\n' +
        '    at Object.getPath (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:55:14)\n' +
        '    at FFmpeg.locate (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:164:33)\n' +
        '    at FFmpeg.spawn (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:144:52)\n' +
        '    at new FFmpeg (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:121:28)\n' +
        '    at createFFmpegStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2951:22)\n' +
        '    at _FFmpegFilterer.createStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:3043:12)\n' +
        '    at #createFFmpegStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2866:50)\n' +
        '    at _GuildQueuePlayerNode.play (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2799:89)\n' +
        '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
        'Error: spawnSync ffmpeg ENOENT\n' +
        '    at Object.spawnSync (node:internal/child_process:1117:20)\n' +
        '    at Object.spawnSync (node:child_process:876:24)\n' +
        '    at FFmpeg.locate (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:165:53)\n' +
        '    at FFmpeg.spawn (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:144:52)\n' +
        '    at new FFmpeg (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:121:28)\n' +
        '    at createFFmpegStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2951:22)\n' +
        '    at _FFmpegFilterer.createStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:3043:12)\n' +
        '    at #createFFmpegStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2866:50)\n' +
        '    at _GuildQueuePlayerNode.play (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2799:89)\n' +
        '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
        'Error: spawnSync avconv ENOENT\n' +
        '    at Object.spawnSync (node:internal/child_process:1117:20)\n' +
        '    at Object.spawnSync (node:child_process:876:24)\n' +
        '    at FFmpeg.locate (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:165:53)\n' +
        '    at FFmpeg.spawn (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:144:52)\n' +
        '    at new FFmpeg (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:121:28)\n' +
        '    at createFFmpegStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2951:22)\n' +
        '    at _FFmpegFilterer.createStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:3043:12)\n' +
        '    at #createFFmpegStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2866:50)\n' +
        '    at _GuildQueuePlayerNode.play (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2799:89)\n' +
        '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
        'Error: spawnSync ./ffmpeg.exe ENOENT\n' +
        '    at Object.spawnSync (node:internal/child_process:1117:20)\n' +
        '    at Object.spawnSync (node:child_process:876:24)\n' +
        '    at FFmpeg.locate (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:165:53)\n' +
        '    at FFmpeg.spawn (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:144:52)\n' +
        '    at new FFmpeg (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:121:28)\n' +
        '    at createFFmpegStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2951:22)\n' +
        '    at _FFmpegFilterer.createStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:3043:12)\n' +
        '    at #createFFmpegStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2866:50)\n' +
        '    at _GuildQueuePlayerNode.play (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2799:89)\n' +
        '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
        'Error: spawnSync ./avconv.exe ENOENT\n' +
        '    at Object.spawnSync (node:internal/child_process:1117:20)\n' +
        '    at Object.spawnSync (node:child_process:876:24)\n' +
        '    at FFmpeg.locate (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:165:53)\n' +
        '    at FFmpeg.spawn (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:144:52)\n' +
        '    at new FFmpeg (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:121:28)\n' +
        '    at createFFmpegStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2951:22)\n' +
        '    at _FFmpegFilterer.createStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:3043:12)\n' +
        '    at #createFFmpegStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2866:50)\n' +
        '    at _GuildQueuePlayerNode.play (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2799:89)\n' +
        '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
        "Error: Cannot find module '@ffmpeg-installer/ffmpeg'\n" +
        'Require stack:\n' +
        '- C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js\n' +
        '- C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-voip\\dist\\index.js\n' +
        '- C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js\n' +
        '- C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\src\\index.js\n' +
        '    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)\n' +
        '    at Module._load (node:internal/modules/cjs/loader:922:27)\n' +
        '    at Module.require (node:internal/modules/cjs/loader:1143:19)\n' +
        '    at require (node:internal/modules/cjs/helpers:119:18)\n' +
        '    at Object.getPath (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:91:19)\n' +
        '    at FFmpeg.locate (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:164:33)\n' +
        '    at FFmpeg.spawn (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:144:52)\n' +
        '    at new FFmpeg (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js:121:28)\n' +
        '    at createFFmpegStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:2951:22)\n' +
        '    at _FFmpegFilterer.createStream (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js:3043:12)\n' +
        "Error: Cannot find module 'ffmpeg-static'\n" +
        'Require stack:\n' +
        '- C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\@discord-player\\ffmpeg\\dist\\index.js\n' +
        '- C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-voip\\dist\\index.js\n' +
        '- C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\node_modules\\discord-player\\dist\\index.js\n' +
        '- C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\mirasaki-music-bot-1.1.4\\src\\index.js\n' +
        '    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)\n' +
        '    at Module._load (node:internal/modules/cjs/loader:922:27)\n' +
        '    at Module.require (node:internal/modules/cjs/loader:1143:19)\n' +
        '    at require (node:internal/modules/cjs/helpers:119:18)\n' +
        '    at Object.getPath (C:\\Users\\Darth GXT\\Downloads\\mirasaki-music-bot-1.1.4\\'... 4385 more characters,
      tts: false,
      nonce: undefined,
      embeds: undefined,
      components: undefined,
      username: undefined,
      avatar_url: undefined,
      allowed_mentions: undefined,
      flags: undefined,
      message_reference: undefined,
      attachments: undefined,
      sticker_ids: undefined,
      thread_name: undefined
    }
  },
  rawError: {
    message: 'Invalid Form Body',
    code: 50035,
    errors: {
      content: {
        _errors: [
          {
            code: 'BASE_TYPE_MAX_LENGTH',
            message: 'Must be 2000 or fewer in length.'
          }
        ]
      }
    }
  },
  code: 50035,
  status: 400,
  method: 'PATCH',
  url:

**Comment by Ariastarcos** [comment]
You don't have ffmpeg installed:
```
Could not locate ffmpeg.
```
Please refer to the documentation: https://github.com/Mirasaki/mirasaki-music-bot?tab=readme-ov-file#installation-and-usage

Download/install links can be found at the provided URL.

**Comment by Ariastarcos** [comment]
From Google Translate:

Vous n'avez pas installé ffmpeg :
```
Impossible de localiser ffmpeg.
```
Veuillez vous référer à la documentation : https://github.com/Mirasaki/mirasaki-music-bot?tab=readme-ov-file#installation-and-usage

Les liens de téléchargement/installation peuvent être trouvés à l’URL fournie.
